go/ethadapter/mgmtcontractlib/mgmt_contract_lib.go

```diff
@@ -118,31 +119,55 @@ func (c *contractLibImpl) DecodeTx(tx *types.Transaction) common.L1TenTransactio
 func (c *contractLibImpl) CreateBlobRollup(t *common.L1RollupTx) (types.TxData, error) {
 	decodedRollup, err := common.DecodeRollup(t.Rollup)
 	if err != nil {
-		panic(err)
+		return nil, fmt.Errorf("failed to decode rollup: %w", err)
+	}
+
+	// Create the blobs here when needed for L1 tx
+	rollupData, err := common.EncodeRollup(decodedRollup)
+	if err != nil {
+		return nil, fmt.Errorf("failed to encode rollup: %w", err)
+	}
+
+	blobs, err := ethadapter.EncodeBlobs(rollupData)
+	if err != nil {
+		return nil, fmt.Errorf("failed to encode rollup to blobs: %w", err)
+	}
+
+	// Verify the blob hash matches what was signed
+	commitment, err := kzg4844.BlobToCommitment(blobs[0])
```

Comment:
there might be a security vulnerability here since we're recreating the blobs that have already been create by the sequencer, might be worth getting @StefanIliev545 to have a look. checking the hash should work but feels wrong. the alternative is that we send the blobs back over GRPC in something like this:

```
struct ExtRollupAndBlobs {
    Rollup *ExtRollup
    Blobs []*kzg4844.Blob
}
```

and then `PublishRollup(producedRollup *common.ExtRollupAndBlobs)`